src/Inputs/InputPassword/InputPassword.tsx

```diff
@@ -1,7 +1,26 @@
-import { InputHTMLAttributes, forwardRef } from 'react'
-import StyledInput from '../styles'
+import { InputHTMLAttributes, forwardRef, useState } from 'react'
+import { StyledInput, StyledToggleInput } from '../styles'
+import { Icon } from '../../Icon'
 
-export const InputPassword = forwardRef<HTMLInputElement, InputHTMLAttributes<HTMLInputElement>>(
-  (props, ref) => <StyledInput type="password" ref={ref} {...props} />,
+export interface PasswordInputProps extends InputHTMLAttributes<HTMLInputElement> {
+  canToggle?: boolean;
+}
+
+export const InputPassword = forwardRef<HTMLInputElement, PasswordInputProps>(({ canToggle = true, ...props}, ref) => {
+
+    const [isVisible, setIsVisible] = useState(false)
+    const resolveIcon = isVisible ? 'visibility_off' : 'visibility'
+    const handleVisible = () => setIsVisible(!isVisible)
+    
+    return (
+      !!canToggle ?
+      <StyledToggleInput>
+        <StyledInput type={isVisible ? '' : 'password'} ref={ref} {...props} />
+        <Icon className="eyeIcon" onClick={handleVisible} icon={resolveIcon}/>
```

Comment:
Small style improvements on the icon.
![image](https://github.com/ynput/ayon-react-components/assets/49156310/0bbcca06-08d0-491c-9ac0-ce4d43a26eab)



```CSS
// remove highlighting
useSelect: none;

// A slight hover effect.
color: var(--md-sys-color-outline);
&:hover {
  color: var(--md-sys-color-on-surface);
}
```

Reply:
Will add.